app/presenters/person_presenter.rb

```diff
@@ -34,14 +31,12 @@ def full_hash_with_profile
   end
 
   def as_json(_options={})
-    attrs = full_hash_with_avatar
-
-    if own_profile? || person_is_following_current_user
-      attrs.merge!(
-        location: @presentable.location,
-        birthday: @presentable.formatted_birthday,
-        bio:      @presentable.bio
-      )
+    attrs = full_hash
+
+    if attrs[:show_profile_info]
+      attrs.merge!(ProfilePresenter.new(profile).private_hash) {|_, v1, _| v1 }
+    else
+      attrs.merge!(ProfilePresenter.new(profile).public_hash) {|_, v1, _| v1 }
     end
```

Comment:
If the cucumber run still passes after the failing specs are adapted,  we should be good.